src/toolkits/feature_engineering/word_counter.cpp

```diff
@@ -125,7 +131,18 @@ flexible_type word_counter_apply_with_manual(const flexible_type& input,
         if (kvp.second.get_type() != flex_type_enum::INTEGER
             && kvp.second.get_type() != flex_type_enum::FLOAT)
           log_and_throw("Invalid type. Dictionary input to WordCounter must have integer or float values.");
-        word_count_delimiters_update(kvp.first.get<flex_string>(), delimiter_list, to_lower, ret_count);
+	if (kvp.second.get_type() == flex_type_enum::INTEGER){
+            word_count_delimiters_update(
+		       kvp.first.get<flex_string>(), delimiter_list, 
+		       to_lower, ret_count, 
+		       run_mode, (float)kvp.second.get<flex_int>());
```

Comment:
why do you need to cast to float? flexible_type can hold an integer value.

Reply:
the logic for `run_mode == INTEGER` and `run_mode == FLOAT` is almost same. I bet flexible<flex_int> can be converted to flexible<flex_float>. You can try to make your logic more precise.